src/components/pages/MyOrganisations/index.tsx

```diff
@@ -1,12 +1,87 @@
 import * as React from 'react';
+import { connect } from 'react-redux';
+import { history } from 'store';
 import styled from '@emotion/styled';
 
+import OrganisationCard from 'components/ui/OrganisationCard';
+import { fetchCurrentUserOrganisations } from 'store/actions/dashboard';
+import { getDateJoinedMessage } from 'helpers/dateHelpers';
+
 import __spacing from 'settings/__spacing';
 
-export default function MyOrganisations(): React.ReactElement<{}> {
-  return <MyOrganisations.Wrapper>My Organisations will go here.</MyOrganisations.Wrapper>;
+interface OrganisationProps {
+  id: string;
+  organisation: {
+    id: string;
+    createdAt: string;
+    name: string;
+    logoUrl: string;
+  };
+  role: {
+    name: string;
+  };
+}
+interface MyOrganisationsProps {
+  img: string;
+  name: string;
+  role: { name: string };
+  date: string;
+  isOrganisationPending: boolean;
+  isOrganizationFetched: boolean;
+  organisations: OrganisationProps;
+  fetchCurrentUserOrganisations: () => void;
 }
 
+export function MyOrganisations(props: MyOrganisationsProps): React.ReactElement<{}> {
+  const { fetchCurrentUserOrganisations, organisations, isOrganisationPending, isOrganizationFetched } = props;
+
+  React.useEffect(function(): void {
+    fetchCurrentUserOrganisations();
+  }, []);
+
+  if (isOrganisationPending) {
+    return <p> ... Loading </p>;
+  } else if (isOrganizationFetched) {
+    const organisationsToDisplay = Object.values(organisations);
+    if (organisationsToDisplay) {
+      return (
+        <MyOrganisations.Wrapper>
+          {organisationsToDisplay.map(org => {
+            return (
+              <div key={org.organisation.id} onClick={() => history.push('/dashboard/profile', org)}>
+                <OrganisationCard
+                  img={org.organisation.logoUrl}
+                  name={org.organisation.name}
+                  role={org.role.name}
+                  date={getDateJoinedMessage(org.organisation.createdAt)}
+                />
+              </div>
+            );
+          })}
+        </MyOrganisations.Wrapper>
+      );
+    }
+  }
+  return <p> Sorry! you do not belong to any organisation yet </p>;
+}
+
+const mapStateToProps = (
+  state,
+):
+  | Pick<MyOrganisationsProps, 'isOrganisationPending'>
+  | Pick<MyOrganisationsProps, 'isOrganizationFetched'>
+  | Pick<MyOrganisationsProps, 'organisations'> => ({
```

Comment:
`Pick<T, K>` is a set of `K` from `T`.
Can you look at the docs for pick [here.](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktk)

Reply:
yea, thanks. Refactoring it asap

Reply:
done